src/test/java/ibm/jceplus/junit/TestMultithread.java

```diff
@@ -50,6 +50,7 @@ public class TestMultithread extends TestCase {
             "ibm.jceplus.junit.openjceplus.multithread.TestDSASignatureInteropSUN",
             "ibm.jceplus.junit.openjceplus.multithread.TestHmacMD5",
             "ibm.jceplus.junit.openjceplus.multithread.TestHmacMD5InteropSunJCE",
+            "ibm.jceplus.junit.openjceplus.multithread.TestMessageDigestClone",
```

Comment:
It seems like this new test runs only SHA-256. I think we want to test cloning on all different MessageDigests that we support. To do this we could create a new `BaseTestMessageDigest` and in this class have it contain the tests you created below for cloning testing. 

For this to work a new constructor would have to be created to set the digest algorithm from parent classes then the variable would have to be used throughout all the base tests and the new  `BaseTestMessageDigest` class. This includes at least `BaseTestMD5, BaseTestSHA1, BaseTestSHA3*, BaseTestSHA224, BaseTestSHA256,BaseTestSHA284, BaseTestSHA512_*, BaseTestSHA512`

Reply:
I added cloning tests for all algorithms with [this](https://github.com/IBM/OpenJCEPlus/pull/98/commits/28c49359b572fdada72a1bfbefd6961500b2b248) commit.